src/main/java/se/curity/oauth/JsonWebKeyType.java

```diff
@@ -41,7 +42,7 @@ static JsonWebKeyType from(JsonValue value)
                     value, value.getValueType()));
         }
 
-        switch (value.toString())
+        switch (((JsonString) value).getString())
```

Comment:
shouldn't polymorphism makes this cast unnecessary?

Reply:
It's calling `getString` not `toString`. `getString` is only available in `JsonString`.